core/eolearn/tests/test_core_tasks.py

```diff
@@ -71,38 +72,39 @@ def patch_fixture():
     return patch
 
 
-def test_copy(patch):
-    patch_copy = CopyTask().execute(patch)
-
-    assert patch == patch_copy, "Copied patch is different"
-
-    patch_copy.data["new"] = np.arange(1).reshape(1, 1, 1, 1)
-    assert "new" not in patch.data, "Dictionary of features was not copied"
+@pytest.mark.parametrize("task", [DeepCopyTask, CopyTask])
+def test_copy(task: CopyTask, patch: EOPatch) -> None:
+    patch_copy = task().execute(patch)
+    assert patch_copy == patch
 
     patch_copy.data["bands"][0, 0, 0, 0] += 1
-    assert np.array_equal(patch.data["bands"], patch_copy.data["bands"]), "Data should not be copied"
-
-
-def test_deepcopy(patch):
-    patch_deepcopy = DeepCopyTask().execute(patch)
+    assert (patch_copy != patch) if task == DeepCopyTask else (patch_copy == patch)
 
-    assert patch == patch_deepcopy, "Deep copied patch is different"
+    patch_copy.data["new"] = np.arange(1).reshape(1, 1, 1, 1)
+    assert "new" not in patch.data
 
-    patch_deepcopy.data["new"] = np.arange(1).reshape(1, 1, 1, 1)
-    assert "new" not in patch.data, "Dictionary of features was not copied"
 
-    patch_deepcopy.data["bands"][0, 0, 0, 0] += 1
-    assert not np.array_equal(patch.data["bands"], patch_deepcopy.data["bands"]), "Data should be copied"
+@pytest.mark.parametrize(
+    "features",
+    [
+        [(FeatureType.MASK_TIMELESS, "mask"), (FeatureType.BBOX, None)],
+        [(FeatureType.TIMESTAMP, None), (FeatureType.SCALAR, "values")],
+    ],
+)
+def test_partial_copy(features: List[FeatureSpec], patch: EOPatch) -> None:
+    patch_copy = DeepCopyTask(features=features)(patch)
```

Comment:
parametrize with the task type (so that CopyTask is also tested) as discussed previous week

Reply:
Updated.

Reply:
another note, the type of `CopyTask` is not `CopyTask` but `Type`. You can use `Type[CopyTask]` to indicate it has to be a `CopyTask` or it's subclass.

Reply:
done